raiden/tests/integration/long_running/test_settlement.py

```diff
@@ -830,3 +835,209 @@ def run_test_automatic_dispute(raiden_network, deposit, token_addresses):
+    identifier = 1
+
+    alice_transfer_secret = sha3(alice_app.raiden.address)
+    alice_transfer_secrethash = sha3(alice_transfer_secret)
+
+    bob_transfer_secret = sha3(bob_app.raiden.address)
+    bob_transfer_secrethash = sha3(bob_transfer_secret)
+
+    hold_event_handler.hold_secretrequest_for(
+        secrethash=alice_transfer_secrethash,
+    )
+    hold_event_handler.hold_secretrequest_for(
+        secrethash=bob_transfer_secrethash,
+    )
+
+    alice_app.raiden.start_mediated_transfer_with_secret(
+        token_network_identifier=token_network_identifier,
+        amount=alice_to_bob_amount,
+        fee=0,
+        target=bob_app.raiden.address,
+        identifier=identifier,
+        secret=alice_transfer_secret,
+    )
+
+    bob_app.raiden.start_mediated_transfer_with_secret(
+        token_network_identifier=token_network_identifier,
+        amount=alice_to_bob_amount,
+        fee=0,
+        target=alice_app.raiden.address,
+        identifier=identifier + 1,
+        secret=bob_transfer_secret,
+    )
+
+    wait_for_state_change(
+        alice_app.raiden,
+        ActionInitTarget,
+        {},
+        retry_timeout,
+    )
+
+    wait_for_state_change(
+        bob_app.raiden,
+        ActionInitTarget,
+        {},
+        retry_timeout,
+    )
+
+    alice_bob_channel_state = get_channelstate(alice_app, bob_app, token_network_identifier)
+    alice_lock = channel.get_lock(alice_bob_channel_state.our_state, alice_transfer_secrethash)
+    bob_lock = channel.get_lock(alice_bob_channel_state.partner_state, bob_transfer_secrethash)
+
+    # This is the current state of protocol:
+    #
+    #    A -> B LockedTransfer
+    #    B -> A SecretRequest
+    #    - protocol didn't continue
+    assert_synced_channel_state(
+        token_network_identifier,
+        alice_app, deposit, [alice_lock],
+        bob_app, deposit, [bob_lock],
+    )
+
+    # A ChannelClose event will be generated, this will be polled by both apps
+    # and each must start a task for calling settle
+    RaidenAPI(bob_app.raiden).channel_close(
+        registry_address,
+        token_address,
+        alice_app.raiden.address,
+    )
+
+    secret_registry_proxy = alice_app.raiden.chain.secret_registry(
+        secret_registry_address,
+    )
+    secret_registry_proxy.register_secret(
+        secret=alice_transfer_secret,
+        given_block_identifier='latest',
+    )
+    secret_registry_proxy = bob_app.raiden.chain.secret_registry(
+        secret_registry_address,
+    )
+    secret_registry_proxy.register_secret(
+        secret=bob_transfer_secret,
+        given_block_identifier='latest',
+    )
+
+    waiting.wait_for_settle(
+        alice_app.raiden,
+        registry_address,
+        token_address,
+        [alice_bob_channel_state.identifier],
+        alice_app.raiden.alarm.sleep_time,
+    )
+
+    alice_app.stop()
+
+    # wait for the node to call batch unlock
+    timeout = 30 if blockchain_type == 'parity' else 10
```

Comment:
Is this something that we can generalize across all our tests? This different in timeout while waiting for a transaction to be mined?

Reply:
can we please not have additional magic constants for the different clients?

Reply:
This was introduced in commit: https://github.com/raiden-network/raiden/commit/dd795be06d6bfad6b82eead4aa26785c48eab044

I assumed it's there for a reason so i copied it into my test similar to `test_batch_unlock`. I removed it but will have to restore it if the test fails. I don't think i should address this in this PR